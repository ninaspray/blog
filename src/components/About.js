import React from 'react';

const About = () => (
    <>
        <h1>About</h1>
        <p>This is my About page</p>
    </>
);

export default About;