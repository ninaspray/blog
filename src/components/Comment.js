import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { CommentType } from '../types/comment.type';

const Comment = ({ comment }) => {
    const [voteCounter, setVoteCounter] = useState(0);
    const [isEmailVisible, setIsEmailVisible] = useState(false);

    const handleClick = event => {
       const shouldIncrease = event.target.value === 'increase';
       const newCounterValue = shouldIncrease
       ? voteCounter + 1
       : voteCounter - 1;
       setVoteCounter(newCounterValue);
    };
    const handleReveal = () => setIsEmailVisible(true);
    const handleHide = () => setIsEmailVisible(false)


    // Display details if:
    // - identity consent was not supported when user posted,
    // - identity consent was supported and user consented
    //   to publish their details.
    // - Adding functionality of the  
    const displayDetails =
        !('identityConsent' in comment) || comment.identityConsent;




    return (
        <li data-testid="comment">
             <div>
                <strong
                    onMouseOver={() => handleReveal()}
                    onFocus={() => handleReveal()}
                    onMouseOut={() => handleHide()}
                    onBlur={() => handleHide()}
                >
                    {displayDetails ? comment.name : 'Anonymous author'}:{' '}
                </strong>
                {displayDetails && isEmailVisible && <div>{comment.email}</div>}
            </div>
            <p>{comment.body}</p>
            {voteCounter}
            <button
                type="button"
                value="increase"
                onClick={event => handleClick(event)}
            >
                + upvote
            </button>
            <button
                type="button"
                value="decrease"
                onClick={event => handleClick(event)}
            >
                - downvote
            </button>
        </li>
    );
};

Comment.propTypes = {
    comment: PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
        body: PropTypes.string,
        email: PropTypes.string,
        identityConsent: PropTypes.bool,
    }).isRequired,
};

export default Comment;