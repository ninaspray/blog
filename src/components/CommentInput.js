import React, { useContext } from 'react';
import PropTypes from 'prop-types';

// custom hook
import { useForm } from '../hooks/useForm';

// contexts
import { ThemeConfig, ThemeContext } from '../contexts/ThemeContext';

const CommentInput = ({ addComment, postId }) => {
    const [input, setInput, resetInput] = useForm({
        name: '',
        email: '',
        body: '',
        postId,
        indentityConsent: false, 
    });
    const { theme } = useContext(ThemeContext);

    const handleSubmit = event => {
        event.preventDefault();
        addComment({ ...input });
        resetInput(event);
    };

    return (
        <form onSubmit={handleSubmit}>
            <div>
                <input
                    style={ThemeConfig[theme]}
                    type="text"
                    name="name"
                    value={input.name}
                    onChange={setInput}
                    placeholder="username"
                />
            </div>
            <div>
                <input
                    style={ThemeConfig[theme]}
                    type="text"
                    name="email"
                    value={input.email}
                    onChange={setInput}
                    placeholder="email"
                />
            </div>
            <div>
                <textarea
                    style={ThemeConfig[theme]}
                    name="body"
                    value={input.body}
                    onChange={setInput}
                    placeholder="comment"
                />
            </div>
            <div>
                <label htmlFor="identityConsent">
                    <input
                        type="checkbox"
                        name="identityConsent"
                        id="identityConsent"
                        onChange={setInput}
                    />
                    Show my name and email
                </label>
            </div>
            <div>
            <button type="submit">Submit</button>
            </div>
        </form>
    );
};

CommentInput.propTypes = {
    addComment: PropTypes.func.isRequired,
    postId: PropTypes.number.isRequired,
};

export default CommentInput;
