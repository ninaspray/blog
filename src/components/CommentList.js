import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import { CommentType } from '../types/comment.type';

const CommentList = ({ comments }) => {
    const [voteCounter, setVoteCounter] = useState(0);
    const handleClick = event => {
        console.log(event.target);
        setVoteCounter(voteCounter + 1);
    };

    useEffect(() => {
        console.log(voteCounter);
    }, [voteCounter]);
    return (
        <ul>
            {comments.map(comment => (
                <li key={comment.id} data-testid="comment">
                    <p>
                    <strong>{comment.name}: </strong>
                        {comment.body}
                    </p>
                    <button type="button" onClick={event => handleClick(event)}>
                        + upvote
                    </button>
                    <button type="button" onClick={event => handleClick(event)}>
                        - downvote
                    </button>
                </li>
            ))}
        </ul>
    );
};

CommentList.propTypes = {
    comments: PropTypes.arrayOf(CommentType).isRequired,
};

export default CommentList;

