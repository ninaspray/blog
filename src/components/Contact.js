import React from 'react';
import { useForm } from '../hooks/useForm';

const Contact = () => {
    const [{name, email, message}, setInput, resetInput] = useForm({
        name: '',
        email: '',
        message: '',
    });

return (
    <form onSubmit={resetInput}>
    <input
    name="name"
    type="text"
    value={name}
    placeholder="Name"
    onChange={setInput}
    />
    <input
    name="email"
    type="email"
    value={email}
    placeholder="e-mail"
    onChange={setInput}
    />
    <input
    name="message"
    type="text"
    value={message}
    placeholder="Message"
    onChange={setInput}
    />
    <button type="submit">submit</button>
    </form>
    );
};

export default Contact;
