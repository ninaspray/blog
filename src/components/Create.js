import React from 'react';
import PropTypes from 'prop-types';

// custom hook
import { useForm } from '../hooks/useForm';

const Create = ({ addPost }) => {
    const [{ title, body }, setInput, resetInput] = useForm({
        title: '',
        body: '',
    });

    const handleSubmit = event => {
        addPost({ title, body });
        resetInput(event);
    };

    return (
        <>
            <h2>Add a new Blog Post</h2>
            <form onSubmit={handleSubmit}>
                <input
                    name="title"
                    type="text"
                    placeholder="Blog Post Title"
                    value={title}
                    onChange={setInput}
                />
                <textarea
                    name="body"
                    placeholder="Blog Post Content"
                    value={body}
                    onChange={setInput}
                />
                <button type="submit">submit</button>
            </form>
        </>
    );
};

Create.propTypes = {
    addPost: PropTypes.func.isRequired,
};

export default Create;