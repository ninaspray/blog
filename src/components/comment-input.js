import React, { useContext } from 'react';
import PropTypes from 'prop-types';

// custom hook
import { useForm } from '../hooks/useFrorm';

// contexts
import { ThemeConfig, ThemeContext } from '../contexts/ThemeContext';

const CommentInput = ({ addComment }) => {
    const [input, setInput, resetInput] = useForm({ username: '', body: '' });
    const { theme } = useContext(ThemeContext);

    const handleSubmit = event => {
        event.preventDefault();
        resetInput();
    };

    return (
        <form onSubmit={handleSubmit}>
            <div>
                <input
                    style={ThemeConfig[theme]}
                    type="text"
                    name="username"
                    value={input.username}
                    onChange={setInput}
                    placeholder="username"
                />
            </div>
            <div>
                <textarea
                    style={ThemeConfig[theme]}
                    name="body"
                    value={input.body}
                    onChange={setInput}
                    placeholder="comment"
                />
            </div>
            <div>
                <button type="submit" onClick={() => addComment(input)}>
                    submit
                </button>
            </div>
        </form>
    );
};

CommentInput.propTypes = {
    addComment: PropTypes.func.isRequired,
};

export default CommentInput;
