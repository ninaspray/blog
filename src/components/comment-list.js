import React from 'react';
import PropTypes from 'prop-types';
import { CommentType } from '../types/comment.type';

const CommentList = ({ comments }) => {
    return (
        <ul>
            {comments.map(comment => (
                <li key={comment.id} data-testid="comment">
                    <p>
                    <strong>{comment.name}: </strong>
                        {comment.body}
                    </p>
                </li>
            ))}
        </ul>
    );
};

CommentList.propTypes = {
    comments: PropTypes.arrayOf(CommentType).isRequired,
};

export default CommentList;

