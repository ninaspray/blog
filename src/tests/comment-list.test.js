import React from 'react';
import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import '../components/comment-list';
import CommentList from '../components/comment-list';

//Adding Test information for user comments
const comments = [
    {
        body: "Test comment body 1",
        username: "Test comment username 1",
        uuid: `Test id 123`,
      },
      {
        body: "Test comment body 2",
        username: "Test comment username 2",
        uuid: "Test id 124",
      },
      {
        body: "Test comment body 3",
        username: "Test comment username 3",
        uuid: "Test id 125",
      }
    ];

describe("comment list", () => {
    it("render expected number of comments", () => {
    const { getAllByTestId } = render(
    <CommentList comments={comments} />
);
  expect(getAllByTestId("comment")).toHaveLength(3);
});
}); 