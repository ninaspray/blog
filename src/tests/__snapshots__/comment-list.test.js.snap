// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comment list renders as expected 1`] = `
<DocumentFragment>
  <ul>
    <li
      data-testid="comment"
    >
      <p>
        <strong>
          Test comment username 1: 
        </strong>
        Test comment body 1
      </p>
    </li>
    <li
      data-testid="comment"
    >
      <p>
        <strong>
          Test comment username 2: 
        </strong>
        Test comment body 2
      </p>
    </li>
    <li
      data-testid="comment"
    >
      <p>
        <strong>
          Test comment username 3: 
        </strong>
        Test comment body 3
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
